packages/blocks/src/submit-button/index.js

```diff
@@ -1,13 +1,22 @@
 /**
  * Internal dependencies
  */
-import { Button } from '../components';
+import { Button, ButtonSpinner } from '../components';
+import { useFormState } from '@crowdsignal/form';
 
 const SubmitButton = ( { attributes, className } ) => {
+	const { isSubmitting } = useFormState();
+
+	const RenderedButton = isSubmitting ? ButtonSpinner : Button;
```

Comment:
I'm not sure I understand... why would we need another `RenderedButton` component instead of just changing the content like:

```
{ isSubmitting ? attributes.label : <Spinner /> }
```

Or basically... I'm not sure there's any reason why the submitting/spinner logic shouldn't be moved inside of this component?

Reply:
To keep the button width while submitting, I'm maintaining the label and just hiding it inside the button. Can you see another way to achieve the same result?

Reply:
Right, but we could just move that logic in here, right? Basically just replace

```
{ attributes.label }
```

With your code from `<RenderedButton>`, so it's something along the lines of:

```
<span className="crowdsignal-forms-submit-button__content">
	{ attributes.label }
</span>
<Spinner />
```